pkg/webhook/certificate_builder_test.go

```diff
@@ -202,7 +204,11 @@ func TestGenCA(t *testing.T) {
 	cb := NewCertificateBuilder(c, log)
 
 	for index, item := range testCases {
-		certs, err := cb.genCA(item.ns, item.svc, item.certPath)
+		certDir, err := ioutil.TempDir("/tmp", item.certPath)
+		if err != nil {
+			t.Errorf("MkdirTemp failed due to %v", err)
+		}
+		certs, err := cb.genCA(item.ns, item.svc, certDir)
 		if err != nil && !item.canMkDir {
 			continue
```

Comment:
I think if err !=nil , it should be raised, otherwise function err will be skipped and unit test is meaningless.

Reply:
I think the error is expectable when canMkDir is false.  FYI @yangyuliufeng